demos/rag_agentic/notebooks/mcp_toolgroup_management.ipynb

```diff
@@ -0,0 +1,372 @@
+    "\n",
+    "- [Remote Setup Guide](https://github.com/opendatahub-io/llama-stack-demos/tree/main/kubernetes)\n",
+    "- [Local Setup Guide](https://github.com/opendatahub-io/llama-stack-demos/blob/main/local_setup_guide.md)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "id": "4401aba8-d618-4b15-abfc-25719e499e3f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from llama_stack_client import LlamaStackClient\n",
+    "from dotenv import load_dotenv\n",
+    "load_dotenv()\n",
+    "\n",
+    "base_url=os.getenv(\"REMOTE_BASE_URL\")\n",
+    "# connecting to remote server\n",
+    "client = LlamaStackClient(base_url=base_url)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2391984d-4e89-4885-80d3-03d7ce7d76ab",
+   "metadata": {},
+   "source": [
+    "## View Registered tool groups"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "374cf724-695f-4340-a813-02f1466b8925",
+   "metadata": {},
+   "source": [
+    "Refer to this [repo](https://github.com/opendatahub-io/llama-stack-demos/tree/main/kubernetes/mcp-servers) to see how different MCP servers are defined and deployed on OpenShift.\n",
+    "\n",
+    "To inspect which toolgroups are currently registered with your LlamaStack server, use:\n",
+    "\n",
+    "```python\n",
+    "client.tools.list()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "328972fd-7338-4555-9be1-b4ee73fe19d4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Registered Toolgroups:\n",
+      " - builtin::code_interpreter\n",
+      " - mcp::github\n",
+      " - mcp::openshift\n",
+      " - builtin::websearch\n",
+      " - builtin::rag\n",
+      " - mcp::slack\n",
+      " - mcp::ansible\n"
+     ]
+    }
+   ],
+   "source": [
+    "toolgroups = client.tools.list()\n",
+    "registered_toolgroups = set(t.toolgroup_id for t in toolgroups)\n",
+    "\n",
+    "print(\"Registered Toolgroups:\")\n",
+    "for tg in registered_toolgroups:\n",
+    "    print(f\" - {tg}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e41f432f-50b2-4a14-9ef0-c6e007c7ddb4",
+   "metadata": {},
+   "source": [
+    "## Register an MCP tool group \n",
+    "\n",
+    "Next, we register a new MCP tool group using `client.toolgroups.register(...)`.\n",
+    "\n",
+    "Ensure that the provided MCP endpoint is valid and accessible."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "125ad514-cf22-41fe-b89e-43203d21bb6b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Register MCP tools\n",
+    "mcp_custom_url = os.getenv(\"CUSTOM_MCP_SERVER_URL\")\n",
+    "client.toolgroups.register(\n",
+    "    toolgroup_id=\"mcp::custom_tool\",\n",
+    "    provider_id=\"model-context-protocol\",\n",
+    "    mcp_endpoint={\"uri\":mcp_custom_url}\n",
+    ")"
```

Comment:
Make sure this is in an if statement so it does not break if the tool is already registered. 
```suggestion
    "# Register MCP tools\n",
    "mcp_custom_url = os.getenv(\"CUSTOM_MCP_SERVER_URL\")\n",
    "if "mcp::custom_tool" not in registered_toolgroups:\n"
    "client.toolgroups.register(\n",
    "    toolgroup_id=\"mcp::custom_tool\",\n",
    "    provider_id=\"model-context-protocol\",\n",
    "    mcp_endpoint={\"uri\":mcp_custom_url}\n",
    ")"
```